waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -818,7 +833,7 @@ proc mountRlnRelayDynamic*(node: WakuNode,
   node.wakuRlnRelay = rlnPeer
 
 
-proc mountRlnRelay*(node: WakuNode, conf: WakuNodeConf) {.raises: [Defect, ValueError, IOError, CatchableError].} =
+proc mountRlnRelay*(node: WakuNode, conf: WakuNodeConf) {.raises: [Defect, ValueError, IOError, CatchableError, Exception].} =
```

Comment:
Where is the `Exception` raised from? Couldn't we find the exact type of Exception being raised and catch it here? Now we have a list of `Defect`, `CatchableError` and `Exception` which just means "everything". I know it's not always easy to catch specific exceptions from submodules. Also see: https://status-im.github.io/nim-style-guide/errors.exceptions.html

Reply:
This has been opened as a new issue. See https://github.com/status-im/nwaku/issues/1045

So far it seems like an OSError but raising it is still giving error that appropriate exception is not raised.

Reply:
There are a few function calls in this proc that can potentially raise an exception, i.e., `readFile`, `parseJson` and 
`to(jsonObject, R andlnMembershipCredentials)`, have you inspected those to see what they raise?

Reply:
Moving all discussions related to Exception handling to the new issue. Review this PR on persistence primarily. @staheri14 Your comment is linked in new issue. https://github.com/status-im/nwaku/issues/1045#issuecomment-1198447379